File: PYTHON/Pertemuan 02/Mahasiswa.py
import sys
sys.stdout.reconfigure(encoding='utf-8')
class Mahasiswa:
    # Konstanta: UPPER_SNAKE_CASE
    IPK_MINIMUM_LULUS = 2.00

    def __init__(self, nama: str, umur: int, ipk: float):
        self._nama = nama
        self._umur = umur
        self._ipk = ipk

    @property
    def nama(self) -> str:
        return self._nama

    @property
    def ipk(self) -> float:
        return self._ipk

    def apakah_lulus(self) -> bool:
        return self._ipk >= self.IPK_MINIMUM_LULUS

    def tampilkan(self):
        print(f"  Nama: {self.nama:<15} | Umur: {self._umur:2d} | IPK: {self.ipk:.2f} | Lulus: {'Ya' if self.apakah_lulus() else 'Tidak'}")

File: JAVA/Pertemuan 02/src/id/ac/uph/pbo/pertemuan02/Mahasiswa.java
package id.ac.uph.pbo.pertemuan02;

public class Mahasiswa {
    // Atribut: camelCase, private (sesuai standar)
    private String nama;
    private int    umur;
    private double ipk;

    // Konstanta: UPPER_SNAKE_CASE
    static final double IPK_MINIMUM_LULUS = 2.00;

    // Konstruktor
    Mahasiswa(String nama, int umur, double ipk) {
        this.nama = nama;
        this.umur = umur;
        this.ipk  = ipk;
    }

    // Method: camelCase
    String getNama()   { return nama; }
    double getIpk()    { return ipk; }
    boolean apakahLulus() { return ipk >= IPK_MINIMUM_LULUS; }

    void tampilkan() {
        System.out.printf("  Nama: %-15s | Umur: %2d | IPK: %.2f | Lulus: %s%n",
                nama, umur, ipk, apakahLulus() ? "Ya" : "Tidak");
    }
}
